src/Composition/object3D.jl

```diff
@@ -632,19 +633,23 @@ hasChildJoint(        obj::Object3D{F}) where F <: Modia3D.VarFloatType = obj.ha
 needsAcceleration(    obj::Object3D{F}) where F <: Modia3D.VarFloatType = obj.computeAcceleration
 objectHasMass(        obj::Object3D{F}) where F <: Modia3D.VarFloatType = obj.hasMass
 isRootObject(         obj::Object3D{F}) where F <: Modia3D.VarFloatType = obj.isRootObj
+isUserDefined(        obj::Object3D{F}) where F <: Modia3D.VarFloatType = obj.isUserDefined
 objectHasMovablePos(  obj::Object3D{F}) where F <: Modia3D.VarFloatType = isdefined(obj.interactionManner, :movablePos)
+isSolid(              obj::Object3D{F}) where F <: Modia3D.VarFloatType = typeof(obj.feature) <: Modia3D.Shapes.Solid{F}
+isVisual(             obj::Object3D{F}) where F <: Modia3D.VarFloatType = typeof(obj.feature) <: Modia3D.Shapes.Visual
 
 featureHasMass(obj::Object3D{F}) where F <: Modia3D.VarFloatType = featureHasMass(obj.feature)
 featureHasMass(feature::Modia3D.AbstractObject3DFeature) = false
 featureHasMass(feature::Modia3D.AbstractScene) = false
 featureHasMass(feature::Shapes.Solid{F}) where F <: Modia3D.VarFloatType = !isnothing(feature.massProperties)
 
-isVisible(obj::Object3D{F}, renderer::Modia3D.AbstractRenderer) where F <: Modia3D.VarFloatType = isVisible(obj.feature, renderer)
-isVisible(obj::Object3D{F}, exportAnimation::Bool) where F <: Modia3D.VarFloatType = isVisible(obj.feature, exportAnimation)
-isVisible(feature::Modia3D.AbstractObject3DFeature, renderer::Modia3D.AbstractRenderer) = false
-isVisible(feature::Modia3D.AbstractObject3DFeature, exportAnimation::Bool) = false
-isVisible(feature::Modia3D.AbstractScene, renderer::Modia3D.AbstractRenderer) = false
-isVisible(feature::Modia3D.AbstractScene, exportAnimation::Bool) = false
+function isVisible(obj::Object3D{F})::Bool where F <: Modia3D.VarFloatType
+    if isSolid(obj) || isVisual(obj)
+        return !isnothing(obj.feature.visualMaterial) && !isnothing(obj.feature.shape)
+    else
+        return false
+    end
+end
```

Comment:
Please, check if a default return type is needed outside of if/else with (code_warntype).

Reply:
'code_warntype' does not report an issue with the explicitly defined return type 'Bool'.